src/test/java/com/google/api/codegen/config/ServiceConfigTest.java

```diff
@@ -0,0 +1,87 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.Authentication;
+import com.google.api.AuthenticationRule;
+import com.google.api.OAuthRequirements;
+import com.google.api.Service;
+import com.google.api.codegen.util.ProtoParser;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class ServiceConfigTest {
```

Comment:
Unfortunately this test is not properly testing the added functionality. It consists of:
- Mocks
- Protobuf objects with some values set (`Service serviceYaml`), plus `ProtoInterfaceModel` (which is also just a set of properties) -  i.e. **test data**.
- But there is no a non-mock functional object (the one, which logic we want to test (protobuf and other POJO objects do not fit, because they are just data, but not the functionality tested by this test)), or a static method, which functionality we want to test. 

Specifically, this PR introduces the following new methods (some of them have actual logic, which is worth to be tested):
- `ProtoApiModel#getAuthScopes()`
- `ProtoApiModel#getAuthScopes(ProtoParser, List<ProtoInterfaceModel>)`
- `DiscoGapicInterfaceContext#getDefaultHost()`
- `GapicInterfaceContext.java#getDefaultHost()`
- `ProtoParser#getDefaultHost()`
- `ProtoParser#getAuthScopes()`

It looks like none of these methods will be called as part of this test execution (the mocks will be, but not the actual methods, i.e. if we check test coverage, the corresponding lines will be reported as untested).